### ISSUE TITLE
Is there a way to disable the up-arrow search history?

### ISSUE THREAD

**Issue by edte** [marlonrichert/zsh-autocomplete#692]
just disable this

**Comment by unode** [comment]
Commenting out https://github.com/marlonrichert/zsh-autocomplete/blob/c7b65508fd3a016dc9cdb410af9ee7806b3f9be1/Functions/Init/.autocomplete__key-bindings#L45 and https://github.com/marlonrichert/zsh-autocomplete/blob/c7b65508fd3a016dc9cdb410af9ee7806b3f9be1/Functions/Init/.autocomplete__key-bindings#L47 seems to do it for me.

That said, I would also see benefit in having a way to disable or customize this behavior as it seems inconsistent with the rest.

**Comment by vedal** [comment]
> That said, I would also see benefit in having a way to disable or customize this behavior as it seems inconsistent with the rest.
For me, thats particularly true when up-arrowing with an empty prompt (just re-run the last command), which I do alot. It would be cool if the search history only showed up after 2 (or N) up-arrows

**Comment by duxbuse** [comment]
Yeah I also wanted to disable it, cause it often locks up for me and I have to kill the terminal. Cant Ctrl+C out of the history menu and then its all buggered.

**Comment by ordy** [comment]
> Yeah I also wanted to disable it, cause it often locks up for me and I have to kill the terminal. Cant Ctrl+C out of the history menu and then its all buggered.

It used to make my terminal freeze too, until I commented the lines mentioned in this comment https://github.com/marlonrichert/zsh-autocomplete/issues/645#issuecomment-1800522781

**Comment by edte** [comment]
This problem has been solved. The requirement is to disable the display of fzf's history when pressing up, but after pressing tab, the arrow keys also can be able to be used in completion. Now the requirements are basically met.

my modifications:


<img width="1877" alt="截屏2024-02-29 11 47 56" src="https://github.com/marlonrichert/zsh-autocomplete/assets/50194671/625bac9f-41da-400a-be34-50f367fcca9e">


pr:
https://github.com/marlonrichert/zsh-autocomplete/compare/main...edte:zsh-autocomplete:main



now:

https://github.com/marlonrichert/zsh-autocomplete/assets/50194671/6af412ac-d891-4e2e-8a03-9e302501ca94

**Comment by unode** [comment]
Could you please leave this open until an actual fix is implemented?
Simply commenting out those lines isn't really a fix.

**Comment by MichaelFlucher** [comment]
I'm also trying to disable the history search. I just want normal scroll through the history like zshnative.
I tested both solutions, commenting out those two key binding lines, and also adding the [function](https://github.com/marlonrichert/zsh-autocomplete?tab=readme-ov-file#reset-history-key-bindings-to-zsh-default) in .zshrc before and after sourcing `zsh-autocomplete.plugin.zsh`. Any suggestions to fix this?

**Comment by jul-sh** [comment]
bumping this, I prefer to default behavior. it would be nice have actual config to support this.

**Comment by zt64** [comment]
Same issue here, tried everything suggested before and none of that works

**Comment by samscientist** [comment]
> Same issue here, tried everything suggested before and none of that works

I commented on the related issue for what worked for me. Give it a try :)
https://github.com/marlonrichert/zsh-autocomplete/issues/645#issuecomment-2131297264

**Comment by jul-sh** [comment]
still unable to disable the arrow up search

**Comment by Dinhero21** [comment]
> the readme states how to reset up/down arrow keys to zsh-default: https://github.com/marlonrichert/zsh-autocomplete?tab=readme-ov-file#reset-history-key-bindings-to-zsh-default

It has since been deleted, "[Restore Zsh-default functionality](https://github.com/marlonrichert/zsh-autocomplete/?tab=readme-ov-file#restore-zsh-default-functionality)" looks really out of place without it

doubt it was intentional

**Comment by MuntasirSZN2** [comment]
> > the readme states how to reset up/down arrow keys to zsh-default: https://github.com/marlonrichert/zsh-autocomplete?tab=readme-ov-file#reset-history-key-bindings-to-zsh-default
> 
> It has since been deleted, "[Restore Zsh-default functionality](https://github.com/marlonrichert/zsh-autocomplete/?tab=readme-ov-file#restore-zsh-default-functionality)" looks really out of place without it
> 
> Edit: [found the commit](https://github.com/marlonrichert/zsh-autocomplete/commit/545051e44407ac42215495d283c37cea9f8be362)

() {
   local -a prefix=( '\e'{\[,O} )
   local -a up=( ${^prefix}A ) down=( ${^prefix}B )
   local key=
   for key in $up[@]; do
      bindkey "$key" up-line-or-history
   done
   for key in $down[@]; do
      bindkey "$key" down-line-or-history
   done
}

All it needs is this.
